doc/tutorials/highlevel.md

```diff
@@ -0,0 +1,322 @@
+
+```c
+ElektraError * error = NULL;
+Elektra * elektra = NULL;
+int rc = loadConfiguration (&elektra, &error);
+
+if (rc == -1)
+{
+    fprintf (stderr, "An error occurred while opening elektra: %s", elektraErrorDescription (error));
+    elektraErrorReset (&error);
+    exit (EXIT_FAILURE);
+}
+
+if (rc == 1)
+{
+    // help mode - application was called with '-h' or '--help'
+    printHelpMessage (elektra, NULL, NULL);
+    elektraClose (elektra);
+    exit (EXIT_SUCCESS);
+}
+```
+
+Next it is recommended, you change the default handler for fatal errors. By default we just call `exit (EXIT_FAILURE)`, since we don't know
+how you log your errors and what cleanup may be needed.
+
+```c
+elektraFatalErrorHandler (elektra, onFatalError);
+```
+
+`onFatalError` will receive the fatal `ElektraError *`. It must at least call `elektraErrorReset` on the error and then call `exit()`.
+
+### Reading config values
+
+Once you have your `Elektra` instance, reading config values is easy. You just call one of the getter functions.
+
+```c
+kdb_double_t mydouble = ELEKTRA_GET (Mydouble) (elektra);
+```
+
+Alternatively, you can use the convenience macro `elektraGet` and/or the tag macros. We recommend you use one of the ways below, to avoid
+problems in case we have to change how we generate the names of the `static inline` functions.
+
+```c
+kdb_double_t mydouble = ELEKTRA_GET (ELEKTRA_TAG_MYDOUBLE) (elektra);
+kdb_double_t mydouble = elektraGet (elektra, ELEKTRA_TAG_MYDOUBLE);
+```
+
+To access config values that don't have static key name, like arrays, you have to supply additional arguments (and use `elektraGetV`):
+
+```c
+kdb_float_t myfloat0 = ELEKTRA_GET (ELEKTRA_TAG_MYFLOATARRAY) (elektra, 0);
+float myfloat1 = ELEKTRA_GET (ELEKTRA_TAG_MYFLOATARRAY) (elektra, 1);
+// or
+kdb_float_t myfloat0 = elektraGetV (elektra, ELEKTRA_TAG_MYFLOATARRAY, 0);
+kdb_float_t myfloat1 = elektraGetV (elektra, ELEKTRA_TAG_MYFLOATARRAY, 1);
+```
+
+Of course we also need to know, how big the `myfloatarray/#` array actually is. To that end we can use `ELEKTRA_SIZE` or `elektraSize`:
+
+```c
+kdb_long_long_t myfloat_size = ELEKTRA_SIZE (ELEKTRA_TAG_MYFLOATARRAY) (elektra);
+// or
+kdb_long_long_t myfloat_size = elektraSize (elektra, ELEKTRA_TAG_MYFLOATARRAY);
+```
+
+Please note that, while it shouldn't happen, if you setup everything correctly, calling a getter on a non-existent, wrongly typed or otherwise
+inconvertible key is a fatal error. All fatal errors result in a call to the fatal error handler and therefore exit the application.
+
+### Writing config values
+
+Writing config values is just as easy as reading them.
+
+```c
+ElektraError * error = NULL;
+ELEKTRA_SET (ELEKTRA_TAG_MYDOUBLE) (elektra, 3.141593, &error);
+if (error == NULL) {
+    // handle error
+    elektraErrorReset (&error);
+}
+```
+
+Admittedly writing is not quite as easy as reading, because of the error handling, but that is just a necessary evil when working in C. It
```

Comment:
Well C **is** the reason that you need to have at least 5 lines of code to make a setter call safe. There are no exceptions and using return values instead of pointer arguments wouldn't change much. But I guess we don't need to mention it here, so I will remove it.

Reply:
Yes, my comment is about if the sentence contributes to the goal of the tutorial, not if it is wrong.